tezos-encoding/src/nom.rs

```diff
@@ -572,6 +573,56 @@ where
     }
 }
 
+#[derive(Debug, PartialEq)]
+/// Lazy, memoized deserialization of a dynamic block.
+pub struct LazyCell<'a, T>
+where
+    T: NomReader<'a>,
```

Comment:
I don't think this requirement is correct - e.g. it will prevent using `LazyCell<Vec<u8>>`, for example.

By all means, `impl<T: NomReader> NomReader for LazyCell<T>` is correct, but requiring it on the struct itself is likely incorrect.

Reply:
I wanted to prevent building a `LazyCell<'a, T>` for a `T` which does not implement `NomReader<'a>` because it would not be possible to `force` it.